universe/engine.py

```diff
@@ -1,110 +1,101 @@
-from decimal import Decimal
-import random
+from . import components, systems
+from .orders import (Move, CargoTransfer, Scrap, BuildInstallation, Terraform,
+                     BuildStation, BuildShip, LaunchMassPacket)
 
 
-class GameState(object):
+class Entity:
+    def __init__(self, registry, data):
+        self._data = data
+        self._components = registry[data['type']]
+
+    def __contains__(self, key):
+        return key in self._components
+
+    def serialize(self):
+        data = {'type': self._data['type']}
+        for _type, component in self._components.items():
+            data.update(component.serialize(self._data))
+        return data
+
+
+class Manager:
+    def __init__(self):
+        self._components = {}
+        self._systems = []
+        self._updates = {}
+
+        self._entity_registry = {}
+
+    def register_system(self, system):
+        self._systems.append(system)
+
+    def register_entity_type(self, name, components):
+        if name in self._entity_registry:
+            raise ValueError("{} is already a registered entity type.".format(name))
+        self._entity_registry[name] = {component.name: component for component in components}
+
+    def get_updates(self, _id):
+        return self._updates.get(_id, [])
+
+    def get_components(self, _type):
```

Comment:
Thoughts: we aren't really getting (or setting) components anymore, but rather references to a whole entity object.  We should probably rename these methods to `get_entities`, `get_entity`, and `set_entity`.